packages/graphql/src/translate/cypher-builder/CypherContext.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CypherParam, CypherReference } from "./references/Reference";
+
+export interface CypherContextInterface {
+    getReferenceId(reference: CypherReference): string;
+    getParamId(reference: CypherParam): string;
+}
```

Comment:
**question:** is there any need for this interface? The old adage of "if an interface has the word 'interface' in it, it probably shouldn't be an interface" comes to mind.

Reply:
This is only to avoid circular dependencies, with some more work we may be able to remove this one, or find a better naming